protostar/migrator/migrator_datetime_state.py

```diff
@@ -0,0 +1,26 @@
+from datetime import datetime as DateTime
+from pathlib import Path
+from typing import Optional
+
+
+class MigratorDateTimeState:
+    def __init__(self, migration_file_path: Path) -> None:
+        self._migration_file_path = migration_file_path
+        self._datetime: Optional[DateTime] = None
+
+    @staticmethod
+    def get_datetime_prefix(datetime: DateTime):
+        return DateTime.strftime(datetime, "%Y%m%d%H%M%S")
+
+    def update_to_now(self) -> None:
+        self._datetime = DateTime.now()
+
+    def get_output_stem(self) -> str:
+        assert self._datetime is not None
+        migration_file_stem = self._migration_file_path.stem
+        prefix = self.get_datetime_prefix(self._datetime)
+        return f"{prefix}_{migration_file_stem}"
+
+    def get_compilation_output_path(self) -> Path:
+        migrations_dir_path = self._migration_file_path.parent
+        return migrations_dir_path / self.get_output_stem()
```

Comment:
I find this abstraction a bit strange. What's the reason for this being a stateful class that has deferred initialization (`update_to_now` method) instead of just being a regular function that takes a path and DateTime object?

Reply:
I need to pass the prefix to the migrator cheatcode(s) and synchronize that prefix between migration history file name and migration output directory.

I don't see how function would solve this problem.

Reply:
If the timestamp of the migration was determined earlier/higher in abstraction, wouldn't it be possible to pass the prefix/path as a stateless data object?

Reply:
discussed privately